src/Models/Scopes/Product/WithProductAttributesScope.php

```diff
@@ -36,7 +36,7 @@ public function apply(Builder $builder, Model $model)
             $attribute = (object) $attribute;
 
             if ($attribute->flat) {
-                if ($attribute->input == 'select' && ! in_array($attribute->source_model, [
+                if ($attribute->input == 'select' && $attribute->type === 'int' && ! in_array($attribute->source_model, [
```

Comment:
Let's do strict comparison (`===`) everywhere, or nowhere to keep code consistent.